tests/SamlSsoTest.php

```diff
@@ -124,4 +128,99 @@ public function create_saml_response_using_samlidp_config(): void
+            'logout' => 'https://anotherfaketest.com',
+            'certificate' => '',
+            'query_params' => false,
+            'encrypt_assertion' => false
+        ];
+
+        // We HAVE to keep the exact format given in the config, i.e (encoded ACS URL => SP configuration)
+        // Otherwise the SamlSso job will not be able to find the correct service provider configuration
+        $encodedAcsUrl = base64_encode($this->fakeACS);
+        config([
+            'samlidp.sp' => [
+                $encodedAcsUrl => $fakeSPConfig
+            ]
+        ]);
+
+        DB::connection()->enableQueryLog();
+
+        // Act
+        $samlResponse = (new SamlSso())->handle();
+
+        // Assert
+        $this->assertEmpty(DB::getQueryLog());
+    }
+
+    /** @test */
+    public function create_response_using_service_provider_model_when_database_access_is_configured(): void
+    {
+        // Arrange
+        config([
+            'samlidp.service_provider_model_usage' => true,
+            'samlidp.service_provider_model' => ServiceProvider::class
+        ]);
+        $serviceProvider = ServiceProvider::factory()->create([
+            'destination_url' => $this->fakeACS // This field HAS to match the ACS URL in the SAML request
+        ]);
+
+        // Act
+        $samlResponse = (new SamlSso())->handle();
+
+        // Assert
+        $this->assertNotNull($samlResponse);
+
+        $expectedFormTag = '<form method="post" action="' .$serviceProvider->destination_url.'">';
+        $this->assertTrue(str_contains($samlResponse, $expectedFormTag));
+    }
+
+    /** @test */
+    public function add_sp_model_configuration_to_existing_service_provider_array_in_config(): void
+    {
+        // Arrange
+        // We need to create an existing service provider configuration in the config so that we can assure that we
+        // dont overwrite it when we add the model's sp configuration to the file
+        config([
+            'samlidp.service_provider_model_usage' => true,
+            'samlidp.service_provider_model' => ServiceProvider::class,
+            'samlidp.sp' => [
+                 'aHR0cHM6Ly9teWZhY2Vib29rd29ya3BsYWNlLmZhY2Vib29rLmNvbS93b3JrL3NhbWwucGhw' => [
+                     'destination' => 'https://myfacebookworkplace.facebook.com/work/saml.php',
+                     'logout' => 'https://myfacebookworkplace.facebook.com/work/sls.php',
+                     'certificate' => '',
+                     'query_params' => false,
+                     'encrypt_assertion' => false
+                    ]
+            ]
+        ]);
+
+        $serviceProvider = ServiceProvider::factory()->create([
+            'destination_url' => $this->fakeACS // This field HAS to match the ACS URL in the SAML request
+        ]);
+
+        // Act
+        $samlResponse = (new SamlSso())->handle();
+
+        // Assert
+        $encodedAcsUrl = base64_encode($serviceProvider->destination_url);
```

Comment:
I would set this variable before setting up the config earlier and use this instead of the magic string on line 194

Reply:
line 192 is a different service provider, I straight up ripped it from the config. The test case is confirming that the model config gets added as well so line 212 is the encoded url for the model